src/jit/codegencommon.cpp

```diff
@@ -4888,6 +4888,82 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
 #pragma warning(pop)
 #endif
 
+
+/*****************************************************************************
+* If any Vector3 args are on stack and they are not pass-by-ref, the upper 32bits
+* must be cleared to zeroes. The native compiler doesn't clear the upper bits
+* and there is no way to know if the caller is native or not. So, the upper
+* 32 bits of Vector argument on stack are always cleared to zero.
+*/
+#ifdef FEATURE_SIMD
+void    CodeGen::genClearStackVec3ArgUpperBits()
+{
+#ifdef DEBUG
+    if (verbose)
+        printf("*************** In genClearStackVec3ArgUpperBits()\n");
+#endif
+
+#if !defined(_TARGET_XARCH_)
+    return;
+#endif // !_TARGET_XARCH_
```

Comment:
This should be `#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING`. I see that below you check for the implicit byref case, but there is no need to even do that checking on Windows.. Also, since you are actually generating xarch code below, you should probably define this in codegenxarch.cpp, and define an empty (at least for now) method on the other targets.